contrib/spouts/kafka/java/heron-kafka-spout-parent/heron-kafka-spout/src/main/java/org/apache/heron/spouts/kafka/KafkaSpout.java

```diff
@@ -275,27 +266,58 @@ public void declareOutputFields(OutputFieldsDeclarer declarer) {
             new Fields(consumerRecordTransformer.getFieldNames(s))));
   }
 
+  @Override
+  public Map<String, Object> getComponentConfiguration() {
+    return null;
+  }
+
+  private void initialize(Map<String, Object> conf) {
+    topologyReliabilityMode = Config.TopologyReliabilityMode.valueOf(
```

Comment:
I am wondering if this valueOf(... toString()) is necessary?

Reply:
unfortunately, it's necessary because the value we put into the config is always a `String`, such as https://github.com/apache/incubator-heron/blob/9f05f47227aa87d0014621519e2d9c89f667a065/heron/api/src/java/org/apache/heron/api/Config.java#L547

And other places in the code base has to do the same thing, such as https://github.com/apache/incubator-heron/blob/9f05f47227aa87d0014621519e2d9c89f667a065/heron/instance/src/java/org/apache/heron/instance/AbstractOutputCollector.java#L71